src/modules/languages/python.nix

```diff
@@ -219,6 +235,11 @@ in
       source ${initVenvScript}
     '') ++ (lib.optional cfg.poetry.install.enable ''
       source ${initPoetryScript}
+    '') ++ (lib.optional cfg.venv.patchelf ''
+      if [ -d "$VIRTUAL_ENV" ]
+      then
+        ${autopatchelf}/bin/autopatchelf --libs ${config.devenv.profile}/lib --path $VIRTUAL_ENV
```

Comment:
I don't know the devenv code well enough but find this approach interesting.
Here a few hints: 1. Avoid running patchelf repeatably as might change the binary on every run (sometimes with weird side effects). 2. Make sure that `${config.devenv.profile}/lib` cannot be garbage collected.

Reply:
Thanks for the feedback!

> 1. Avoid running patchelf repeatably as might change the binary on every run (sometimes with weird side effects).

Indeed. Maybe storing the profile inside `VIRTUAL_ENV` and comparing it is a good enough approach.

> 2. Make sure that ${config.devenv.profile}/lib cannot be garbage collected.

I presumed the profile is not GCed, as it has a GC root that in `.devenv`. Not entirely sure whether this is a good presumption though.